data/core/view.lua

```diff
@@ -81,6 +81,7 @@ function View:on_mouse_pressed(button, x, y, clicks)
     self.dragging_scrollbar = true
     return true
   end
+  return keymap.on_key_pressed(button:sub(1,1) .. "click", x, y, clicks)
```

Comment:
I think it would be better to do something like this in `core.on_event`:
```lua
elseif type == "mousepressed" then
    did_keymap = keymap.on_key_pressed(...) -- TODO: Translate the event name appropriately
    if not did_keymap then
        core.root_view:on_mouse_pressed(...)
    end
```
Also we should make `src/api/system.c:button_name` return values for other mouse buttons, as now we just return `?`.

Do we need to do this also for `mousereleased`?

Reply:
> I think it would be better to do something like this in core.on_event:

Yeah, agreed, I'll change it. I originally did it in view, because I wanted to make the views easily able to override this new behaviour, but we should keep the same paradigm across keymaps in the long run.

> Also we should make src/api/system.c:button_name return values for other mouse buttons, as now we just return ?.

Sure, I'll add in X1 and X2?

> Do we need to do this also for mousereleased?

No. Keymap doesn't do this, so let's keep it as keymap is.